### ISSUE TITLE
null is not an object (evaluating 'child.props')

### ISSUE THREAD

**Issue by snakeyak** [Lesha-spr/react-validation#43]
Hi,

I’ve been stuck on this error for over a day now. When I use react-validation v2, I get this error:
```
[Error] TypeError: null is not an object (evaluating 'child.props')
[Error] Warning: performUpdateIfNecessary: Unexpected batch number (current 8, pending 3)
```

And it is pointing to this line:
https://github.com/Lesha-spr/react-validation/blob/master/src/components/Form.react.jsx#L106

I’ve tracked it down to a piece of my code, that dynamically adds/ removes markup based on props, within <Validation.components.Form, e.g.

```
<Validation.components.Form onSubmit={this.onContinueButtonClick.bind(this)}>
…
…
…
    {this.props.deliveryType != 't' && this.props.deliveryType != 'a' &&
      <div className="row bottom10">
        <div className="col-xs-12">
          <input
            type="checkbox"
            onChange={(e) => { if (!e) return false; this.sameAddressChecbox(e.target.checked); }}
          />
          Same as delivery address
        </div>
      </div>
    }

    {!this.state.billingAddressDisabled &&
      <AddressFinder onAddressSelected={this.addressSelected.bind(this)} />
    }
…
…
…
</Validation.components.Form>
```

When I remove that bit it doesn’t produce the error anymore. I can't tell whether I'm doing something wrong or whether this is a bug within react-validation. However when I use v1.7.5. with the same code, everything works fine.

Any help please?

**Comment by Cyval** [comment]
I've also encountered this one, wherein providing other props other than react-validation component causes it to produce an error.

I have a strong feeling that you should provide only react-validation components, e.g. `Input`, `Select`, . . .etc' for that to be fixed. This can be an issue though, since some developers tend to put props inside the form for some kind of component validation.

Upon encountering this one, what I did was:
1. Put the props outside the form, or 
2. Put the validation of props via state. 

I also hope a workaround can be made for this one.

**Comment by MattAttack07** [comment]
I am also seeing the same issue when nesting the Validation.components.Input in a form-group class. It also causes errors when there is a separate label component.

`<div className="form-group">
    <label htmlFor="institution_name">Name</label>
    <Validation.components.Input
       ref="name"
       blocking="input"
       className='form-control'
       placeholder="Sandbox"
       validations={['required']}
       invalidClassName=name_form_error'
       name='name'
       type='text'/>
 </div>`

**Comment by Cyval** [comment]
The problem above, as I have experienced, is that every time that an `Input` component is found by the library, it `registers` each component to the form `Validation.components.Form` component. So I have a feeling that they're attaching the input components inside the form wrapper since the form has something to do with validation as a whole.

**Comment by Lesha-spr** [comment]
Hi @snakeyak ,

Sorry for late fix. I've commited 0b1c95bb4ce4de9d260b9a7d2e9fd934534acdad which fixes null rendering (and it covers your issue).

Please check out 2.0.5 for fix.

**Comment by Lesha-spr** [comment]
Hi @MattAttack07 ,

You shouldn't use validation components outside the ```Validation.components.Form``` component.

Please try to wrap it and share your results.

**Comment by Lesha-spr** [comment]
Hi @Cyval ,

I'm not sure I've understood you correctly. Could you please provide an example of an issue you are facing on?

Thanks,
Oleksii

**Comment by Cyval** [comment]
Sure. It's like a server-side validation for forms

http://pastebin.com/ewmAZv73

**Comment by Lesha-spr** [comment]
Hi @Cyval ,

2.0.5 should fix your issue too. Please check it out. Also feel free to apply PR if it doesn't resolve your issue.

Thanks,
Oleksii

**Comment by MattAttack07** [comment]
@Lesha-spr this was a single element from my form. All elements are wrapped in `Validation.components.Form`, and fails to load as it tries to find the validations on the label, and form-group classes. Producing an error in console `TypeError: null is not an object`

**Comment by snakeyak** [comment]
this works for me now, thank you!
